opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -55,3 +58,47 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+
+
+class ResourceDetector:
+    def __init__(self, crash_on_error=False):
+        self.crash_on_error = crash_on_error
+
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        return _EMPTY_RESOURCE
+
+
+class OTELResourceDetector:
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        env_resources_items = os.environ.get("OTEL_RESOURCE")
+        env_resource_map = {}
+        if env_resources_items:
+            for item in env_resources_items.split(","):
+                key, value = item.split("=")
+                env_resource_map[key.strip()] = value.strip()
```

Comment:
I guess this is a matter of taste or coding style, but I tend to find dict comprehensions more readable than dict building:
```
env_resource_map = {
    key.strip(): value.strip()
        for key,value in (item.split('=') for item in env_resources_items.split(','))
}
```